bundle/edu.gemini.catalog/src/main/scala/edu/gemini/catalog/api/CatalogQuery.scala

```diff
@@ -19,8 +19,10 @@ case class MagnitudeQueryFilter(mc: MagnitudeConstraints) extends QueryResultsFi
 
 sealed abstract class CatalogName(val id: String, val displayName: String) {
   def supportedBands: List[MagnitudeBand] = Nil
-  // Indicates what is the band used when a generic R band is requried
+  // Indicates what is the band used when a generic R band is required
   def rBand: MagnitudeBand = MagnitudeBand.UC
+  // Whether we should validate the xml on parsing
+  def checkValidity: Boolean = true
```

Comment:
I considered that, it would be possible to create a "pre-check" method that does the validation check. I went against that because the most logical place to do that is on the `CatalogAdapters`.

However those are created at runtime based on the content of an already validated xml producing  a chicken an egg problem